src/index.ts

```diff
@@ -1 +1,64 @@
-console.log('hey!');
+import got from 'got';
+
+import { reject, isNil, join } from 'ramda';
+
+import { ISale } from './types';
+
+const withoutUndefined = reject(isNil);
+const apiUrl = 'https://app.enotas.com.br/api';
+
+interface ISalesParams {
+  readonly filter: string;
+  readonly pageNumber: number;
+  readonly pageSize: number;
+  readonly orderBy: number;
+}
+
+const enotas = ({ apiKey }: { readonly apiKey: string } = { apiKey: '' }) => {
+  if (!apiKey) {
+    throw new Error('missing parameter "apiKey"');
+  }
+
+  const gotClient = got.extend({
+    baseUrl: apiUrl,
+    headers: {
+      Authorization: `Basic ${apiKey}`
+    },
+    json: true
+  });
+
+  return {
+    getSales: ({ filter, pageNumber, pageSize, orderBy }: ISalesParams) => {
+      const query = withoutUndefined({
+        filter,
+        pageNumber,
+        pageSize,
+        orderBy
+      });
+
+      return gotClient.get('/vendas/getFilterBy', {
+        query
+      });
+    },
+    createSale: (sale: ISale) => {
+      return gotClient.post('/vendas', {
+        body: mapSale(sale)
+      });
+    }
+  };
+};
+
+const joinWithSemicolon = join(';');
```

Comment:
Seems to be a util

Reply:
Since we're not using it in other files (yet) I didn't bother moving it to a "utils" file or something like that. But we can do it.

Reply:
I think that we will build it faster. I open 2 PR's isolated of things like Test / Lint, and you add test stuff inside your PR's. So, the change of a PR replicates the same function for we did not segregate it, will be big (in my opinion). If you follow my change suggestions I think that separate it will be the smallest one.